utils/configuration.go

```diff
@@ -81,37 +84,72 @@ func ParseLogLevel(configuration *viper.Viper, defaultLevel logrus.Level) (
 	return logrus.ParseLevel(logStr)
 }
 
-// ParseStorage tries to parse out Storage from a Viper.  If backend and
+// ParseSQLStorage tries to parse out Storage from a Viper.  If backend and
 // URL are not provided, returns a nil pointer.  Storage is required (if
 // a backend is not provided, an error will be returned.)
-func ParseStorage(configuration *viper.Viper, allowedBackends []string) (*Storage, error) {
+func ParseSQLStorage(configuration *viper.Viper) (*Storage, error) {
 	store := Storage{
 		Backend: configuration.GetString("storage.backend"),
 		Source:  configuration.GetString("storage.db_url"),
 	}
 
-	supported := false
-	store.Backend = strings.ToLower(store.Backend)
-	for _, backend := range allowedBackends {
-		if backend == store.Backend {
-			supported = true
-			break
-		}
-	}
-
-	if !supported {
+	switch {
+	case store.Backend != notary.MySQLBackend && store.Backend != notary.SQLiteBackend:
+		return nil, fmt.Errorf(
+			"%s is not a supported SQL backend driver",
+			store.Backend,
+		)
+	case store.Source == "":
 		return nil, fmt.Errorf(
-			"must specify one of these supported backends: %s",
-			strings.Join(allowedBackends, ", "))
+			"must provide a non-empty database source for %s",
+			store.Backend,
+		)
 	}
+	return &store, nil
+}
 
-	if store.Backend == MemoryBackend {
-		return &Storage{Backend: MemoryBackend}, nil
+// ParseRethinkDBStorage tries to parse out Storage from a Viper.  If backend and
+// URL are not provided, returns a nil pointer.  Storage is required (if
+// a backend is not provided, an error will be returned.)
+func ParseRethinkDBStorage(configuration *viper.Viper) (*RethinkDBStorage, error) {
+	store := RethinkDBStorage{
+		Storage: Storage{
+			Backend: configuration.GetString("storage.backend"),
+			Source:  configuration.GetString("storage.db_url"),
+		},
+		CA:      configuration.GetString("storage.tls_ca_file"),
```

Comment:
Agree for both!  I'll change it to a relative path for CA unless there are any objections from @endophage